crml/cennzx/rpc/src/lib.rs

```diff
@@ -160,47 +285,54 @@ where
 			data: Some(format!("{:?}", e).into()),
 		})?;
 
-		let liquidity = TryInto::<u64>::try_into(result.0.saturated_into::<u128>()).map_err(|e| RpcError {
-			code: ErrorCode::ServerError(Error::PriceOverflow.into()),
-			message: "Liquidity too large.".into(),
-			data: Some(format!("{:?}", e).into()),
-		})?;
-		let core = TryInto::<u64>::try_into(result.1.saturated_into::<u128>()).map_err(|e| RpcError {
-			code: ErrorCode::ServerError(Error::PriceOverflow.into()),
-			message: "Core asset too large.".into(),
-			data: Some(format!("{:?}", e).into()),
-		})?;
-		let asset = TryInto::<u64>::try_into(result.2.saturated_into::<u128>()).map_err(|e| RpcError {
-			code: ErrorCode::ServerError(Error::PriceOverflow.into()),
-			message: "Trade asset too large.".into(),
-			data: Some(format!("{:?}", e).into()),
-		})?;
-		Ok((liquidity, core, asset))
+		Ok(LiquidityValueResponse {
+			liquidity: result.0,
+			core: result.1,
+			asset: result.2,
+		})
 	}
 
-	fn liquidity_price(&self, asset_id: AssetId, liquidity_to_buy: Balance) -> Result<(u64, u64)> {
+	fn liquidity_price(
+		&self,
+		asset_id: AssetId,
+		liquidity_to_buy: WrappedBalance,
+	) -> Result<LiquidityPriceResponse<Balance>> {
 		let api = self.client.runtime_api();
 		let best = self.client.info().best_hash;
 		let at = BlockId::hash(best);
 
 		let result = api
-			.liquidity_price(&at, asset_id, liquidity_to_buy)
+			.liquidity_price(&at, asset_id, liquidity_to_buy.0.into())
 			.map_err(|e| RpcError {
 				code: ErrorCode::ServerError(Error::Runtime.into()),
 				message: "Unable to query liquidity price.".into(),
 				data: Some(format!("{:?}", e).into()),
 			})?;
 
-		let core = TryInto::<u64>::try_into(result.0.saturated_into::<u128>()).map_err(|e| RpcError {
-			code: ErrorCode::ServerError(Error::PriceOverflow.into()),
-			message: "Core asset too large.".into(),
-			data: Some(format!("{:?}", e).into()),
-		})?;
-		let asset = TryInto::<u64>::try_into(result.1.saturated_into::<u128>()).map_err(|e| RpcError {
-			code: ErrorCode::ServerError(Error::PriceOverflow.into()),
-			message: "Trade asset too large.".into(),
-			data: Some(format!("{:?}", e).into()),
-		})?;
-		Ok((core, asset))
+		Ok(LiquidityPriceResponse {
+			core: result.0,
+			asset: result.1,
+		})
 	}
 }
+
+#[test]
+fn wrapped_balance_can_deserialize_integer_or_hex() {
+	let info = WrappedBalance { 0: u64::MAX.into() };
```

Comment:
should use this syntax for tuple struct
```suggestion
	let info = WrappedBalance(u64::MAX.into());
```